services/category.js

```diff
@@ -0,0 +1,21 @@
+const Category = require('~/models/category')
+const { createError } = require('~/utils/errorsHelper')
+const { CATEGORY_NOT_FOUND } = require('~/consts/errors')
+
+const categoryService = {
+  getCategories: async (searchFilter, skip, limit) => {
+    return await Category.find(searchFilter).populate('subjects').skip(skip).limit(limit).lean().exec()
```

Comment:
i am pretty sure that virtuals don't work with lean()

Reply:
https://stackoverflow.com/questions/18363472/mongoose-lean-query-virtuals-not-showing

Reply:
I've tested it and everything worked fine

Reply:
weird, didn't work for me